src/main/java/de/maxhenkel/audioplayer/command/AudioPlayerCommands.java

```diff
@@ -426,10 +415,19 @@ private static LiteralArgumentBuilder<CommandSourceStack> bulkApplyCommand(Liter
                                     ItemStack itemInHand = player.getItemInHand(InteractionHand.MAIN_HAND);
                                     String customName = StringArgumentType.getString(context, "custom_name");
                                     if (validator.test(itemInHand)) {
-                                        // renameItem(context, itemInHand, sound, customName);
-                                        // Will copy solution into this section once complete.
+                                        ListTag shulkerContents = itemInHand.getTagElement("BlockEntityTag").getList("Items",10);
+                                        for (int i = 0; i < shulkerContents.size(); i++) {
+                                            CompoundTag currentItem = shulkerContents.getCompound(i);
+                                            ItemStack itemStack = ItemStack.of(currentItem);
+                                            if (itemStack.getItem() instanceof RecordItem) {
+                                                renameItem(context, itemStack, sound, customName);
+                                                shulkerContents.getCompound(i).put("tag", itemStack.getTag());
+                                            }
+                                        }
+                                        itemInHand.getTagElement("BlockEntityTag").put("Items", shulkerContents);
+                                        context.getSource().sendSuccess(Component.literal("Successfully updated Shulker Box contents"), false);
                                     } else {
-                                        context.getSource().sendFailure(Component.literal("You don't have a %s in your main hand".formatted(itemTypeName)));
+                                        context.getSource().sendFailure(Component.literal("You don't have a Shulker Box in your main hand"));
                                     }
                                     return 1;
```

Comment:
This is duplicate code. If you have the same logic multiple times, please extract the code into a method.